### ISSUE TITLE
Rule proposal: `no-error-prototype-property-assign`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1883]
### Description

Forbid assign `name`, `stack` (maybe also `cause`) to error.
It's common to assign values to error for debug.

```js
throw Object.assign(new Error('message'), {someUsefulInfomationCantDisplayInMessage})
```

Normally it's fine, but if you happed have `name`, `stack`, this may break the error.
I happed use both [`name`](https://github.com/prettier/prettier/blob/e2337dce26e83d3604fb7a9be7e3e58311fec2b5/src/main/create-print-pre-check-function.js#L53) and [`stack`](https://github.com/prettier/prettier/blob/e2337dce26e83d3604fb7a9be7e3e58311fec2b5/src/main/create-print-pre-check-function.js#L55) in one place recently.

### Fail

```js
throw Object.assign(error, {name})
```

```js
throw Object.assign(error, {stack})
```

### Pass

```js
throw Object.assign(error, {notNameNotStack})
```

**Comment by silverwind** [comment]
Would it detect regular assignments as well? e.g. `error.name = 'foo'`

**Comment by fregante** [comment]
Should this also cover error subclasses? If not, why?

I also hate properties on errors because they're invisible in logs, but they're commonly used (maybe also because any `catch` can and does use those properties)

I kinda like the rule, but also think that they can be useful.
